test/behaviour/typeql/TypeQLSteps.java

```diff
@@ -181,12 +181,14 @@ private void assertQueryPlansCorrect(TypeQLMatch typeQLQuery) {
                 .toSet();
         for (Conjunction conjunction : disjunction.conjunctions()) {
             GraphTraversal.Thing traversal = conjunction.traversal(filter);
-            // limited permutation space to avoid OOMs and timeouts
-            FunctionalIterator<GraphProcedure> procedurePermutations = ProcedurePermutator.generate(traversal.structure()).limit(5000);
-            Set<VertexMap> answers = procedurePermutations.next().iterator(tx().concepts().graph(),
+            // limited permutation space to avoid timeouts
+            FunctionalIterator<GraphProcedure> procedurePermutations = ProcedurePermutator.generate(traversal.structure()).limit(20000);
+            GraphProcedure procedure = procedurePermutations.next();
```

Comment:
introduce a procedure variable so we can easily breakpoint and investigate!

Reply:
Why the limit increase?

Reply:
Mostly because we can! They still run fast for all the behaviour scenarios.

However, now that I'm thinking of it, we should choose a factorial, like 8! = 40320 to ensure we can get all query plans for queries with that many traversal vertices?